### ISSUE TITLE
thread '<main>' panicked at 'called when calling 'trace_transaction'

### ISSUE THREAD

**Issue by mtbitcoin** [ethcore/parity#1704]
Attempting to call trace_transaction causes an exception in the latest Parity/v1.3.0-unstable-a76981a-20160724/x86_64 as below:

C:\Users\MT4\.cargo\bin>parity --tracing on
2016-07-24 09:54:18 Starting Parity/v1.3.0-unstable-a76981a-20160724/x86_64-windows-msvc/rustc1.10.0
2016-07-24 09:54:18 Using state DB journalling strategy fast
2016-07-24 09:54:18 Configured for Frontier/Homestead using Ethash engine
2016-07-24 09:54:18 Public node URL: enode://7c81c7305c075e81399897e500099565e733cfb67e7c2034c7b52fc73b769bd832f6c29d31c07541c57d083b8d1f01228a01257fa794c87bea36a7c2bf99a771@127.0.0.1:30303
2016-07-24 09:54:49     0/ 3/25 peers     37 MiB db   10 KiB chain    2 KiB queue   10 KiB sync
2016-07-24 09:55:19     0/ 3/25 peers     37 MiB db   10 KiB chain    2 KiB queue   10 KiB sync
2016-07-24 09:55:50     0/ 3/25 peers     37 MiB db   10 KiB chain    2 KiB queue   10 KiB sync
2016-07-24 09:56:20     0/ 3/25 peers     37 MiB db   12 KiB chain    2 KiB queue   10 KiB sync
thread '<unnamed>' panicked at 'DecodeError: RlpExpectedToBeList', Users\MT4\.cargo\git\checkouts\parity-6ba620efeffb2216\master\util\src\rlp\rlpin.rs:119
note: Run with `RUST_BACKTRACE=1` for a backtrace.
2016-07-24 09:56:29 Finishing work, please wait...
thread '<main>' panicked at 'called `Result::unwrap()` on an `Err` value: Closed', ../src/libcore\result.rs:785

**Comment by gavofyork** [comment]
could you provide further details on the arguments to `trace_transaction`?

**Comment by arkpar** [comment]
DB requires a resync after #1698 I guess

**Comment by mtbitcoin** [comment]
@gavofyork  

curl http://localhost:8546 -X POST --data "{\"jsonrpc\":\"2.0\",\"method\":\"trace_transaction\",\"params\":[\"0xe560ba51316d9224ad424119e2d2d647122958314a0b52bd586935188f024a3b\"],\"id\":1}" -H "Content-Type: application/json" >

**Comment by johnkozan** [comment]
I can confirm this.  The following RPC call will also crash parity (backtrace below).

```bash
curl -X POST -H "Content-type: application/json" --data '{"jsonrpc":"2.0","method":"trace_transaction","params":["0x3c29cfb1c194db3034b30876aeb291706dff89f0ff2eed8f247ee25ddcc073aa"],"id":63343}'  localhost:8545
```


Using parity from git master:
```
 version Parity/v1.3.0-unstable-8574bfd-20160724/x86_64-linux-gnu/rustc1.12.0-nightly  
```

Parity was run with the --tracing on flag, and the DB was deleted and freshly re-synced.

During the initial sync, parity crashes with the following error.  I restart parity and the sync continues as normal.  This crash is also reproducible. 
```
2016-07-25 21:22:21 Syncing #1430475 ec0dâ€¦a815    210 blk/s 1280 tx/s  31 Mgas/s       0+29882 Qed   #1460399    6/26/25 peers     32 MiB db    2 MiB chain   25 MiB queue    1 MiB sync

thread 'IO Worker #0' has overflowed its stack
fatal runtime error: stack overflow
Aborted (core dumped)
```

My backtrace:
```
thread '<unnamed>' panicked at 'DecodeError: RlpIsTooShort', util/src/rlp/rlpin.rs:119
stack backtrace:
1:     0x55fbaaee012f - std::sys::backtrace::tracing::imp::write::h29f5fdb9fc0a7395
2:     0x55fbaaeea2eb - std::panicking::default_hook::_{{closure}}::h2cc84f0378700526
3:     0x55fbaaee7c3a - std::panicking::default_hook::hbbe7fa36a995aca0
4:     0x55fbaaee828c - std::panicking::rust_panic_with_hook::h105c3d42fcd2fb5e
5:     0x55fbaaee80e1 - std::panicking::begin_panic::hbf62ea4a5ff3f9de
6:     0x55fbaaee800a - std::panicking::begin_panic_fmt::h20f5943904e5791d
7:     0x55fbaaabd920 - _<ethcore_util..kvdb..Database as ethcore..db..Readable>::read::_{{closure}}::h601b7564885109ad
at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libcore/result.rs:695
at /home/john/code/parity/util/src/rlp/rlpin.rs:119
at /home/john/code/parity/util/src/rlp/rlpin.rs:124
at /home/john/code/parity/util/src/rlp/mod.rs:90
at /home/john/code/parity/ethcore/src/db.rs:164
8:     0x55fbaaa17f43 - _<ethcore..trace..db..TraceDB<T>>::traces::h5a7172974d0ddb60
at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libcore/option.rs:385
at /home/john/code/parity/ethcore/src/db.rs:164
at /home/john/code/parity/ethcore/src/db.rs:124
at /home/john/code/parity/ethcore/src/trace/db.rs:157
9:     0x55fbaaac1311 - _<ethcore..trace..db..TraceDB<T> as ethcore..trace..Database>::transaction_traces::_{{closure}}::hb25808a6767ee5b1
at /home/john/code/parity/ethcore/src/trace/db.rs:162
at /home/john/code/parity/ethcore/src/trace/db.rs:292
10:     0x55fbaaa0a1bc - _<ethcore..client..client..Client as ethcore..client..traits..BlockChainClient>::transaction_traces::ha649a52214094533
at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libcore/option.rs:572
at /home/john/code/parity/ethcore/src/trace/db.rs:291
at /home/john/code/parity/ethcore/src/client/client.rs:933
at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libcore/option.rs:572
at /home/john/code/parity/ethcore/src/client/client.rs:932
at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libcore/option.rs:572
at /home/john/code/parity/ethcore/src/client/client.rs:930
11:     0x55fbaa6fe9bc - _<ethcore_rpc..v1..impls..traces..TracesClient<C, M> as ethcore_rpc..v1..traits..traces..Traces>::transaction_traces::_{{closure}}::hc4145ca26a2d5174
12:     0x55fbaa5f54e2 - _<jsonrpc_core..io..DelegateMethod<T, F> as jsonrpc_core..commander..MethodCommand>::execute::h837cf2c7e9c66039
2016-07-25 19:11:39 Imported #1953006 50f4â€¦98d7 (4 txs, 0.21 Mgas, 3.59 ms, 1.03 KiB)
2016-07-25 19:11:42     0/22/25 peers     49 MiB db    2 MiB chain    2 KiB queue   10 KiB sync
at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libcore/result.rs:590
at /home/john/code/parity/rpc/src/v1/impls/traces.rs:92
at /home/john/code/parity/target/release/build/jsonrpc-core-ab6daab4ab051727/out/lib.rs:2176
13:     0x55fbaab68323 - jsonrpc_core::commander::Commander::execute_method::h292ac5972ac7e8d3
at /home/john/code/parity/target/release/build/jsonrpc-core-ab6daab4ab051727/out/lib.rs:2062
14:     0x55fbaab69aa9 - jsonrpc_core::request_handler::RequestHandler::handle_call::h005546ba98ba7bb7
at /home/john/code/parity/target/release/build/jsonrpc-core-ab6daab4ab051727/out/lib.rs:2138
at /home/john/code/parity/target/release/build/jsonrpc-core-ab6daab4ab051727/out/lib.rs:2123
15:     0x55fbaab68e4f - jsonrpc_core::request_handler::RequestHandler::handle_request::h9d44ce8465bd5326
at /home/john/code/parity/target/release/build/jsonrpc-core-ab6daab4ab051727/out/lib.rs:2108
16:     0x55fbaab6d15f - jsonrpc_core::io::IoHandler::handle_request::h298ee2af773eb807
at /home/john/code/parity/target/release/build/jsonrpc-core-ab6daab4ab051727/out/lib.rs:2277
17:     0x55fbaab172d3 - _<jsonrpc_http_server..handler..ServerHandler as hyper..server..Handler<hyper..net..HttpStream>>::on_request_readable::h48dc43e765bd9af1
at /home/john/.cargo/git/checkouts/jsonrpc-http-server-67dcc044a40796d0/master/src/handler.rs:130
18:     0x55fbaab0f1c8 - _<hyper..http..conn..ConnInner<K, T, H>>::read::h59f7745d49180dac
at /home/john/.cargo/git/checkouts/hyper-03790fdbcffbe388/master/src/server/message.rs:38
at /home/john/.cargo/git/checkouts/hyper-03790fdbcffbe388/master/src/http/conn.rs:266
19:     0x55fbaab0f741 - _<hyper..http..conn..ConnInner<K, T, H>>::read::h59f7745d49180dac
at /home/john/.cargo/git/checkouts/hyper-03790fdbcffbe388/master/src/http/conn.rs:285
20:     0x55fbaab0fd05 - _<hyper..http..conn..ConnInner<K, T, H>>::read::h59f7745d49180dac
at /home/john/.cargo/git/checkouts/hyper-03790fdbcffbe388/master/src/http/conn.rs:171
21:     0x55fbaab0cdf3 - _<hyper..http..conn..ConnInner<K, T, H>>::on_readable::heff2cea519929730
at /home/john/.cargo/git/checkouts/hyper-03790fdbcffbe388/master/src/http/conn.rs:449
22:     0x55fbaab097d8 - _<hyper..http..conn..Conn<K, T, H>>::ready::hce7404c1882f333f
at /home/john/.cargo/git/checkouts/hyper-03790fdbcffbe388/master/src/http/conn.rs:530
23:     0x55fbaab26880 - _<rotor..handler..Handler<M> as mio..handler..Handler>::ready::_{{closure}}::h7601176c8f1285a1
24:     0x55fbaab2281b - rotor::handler::machine_loop::_{{closure}}::h405b82e8a5c9f314
25:     0x55fbaaafea0b - _<mio..event_loop..EventLoop<H>>::run_once::h434850a2c235de76
at /home/john/.cargo/registry/src/github.com-1ecc6299db9ec823/slab-0.1.3/src/lib.rs:186
at /home/john/.cargo/git/checkouts/rotor-d280b489cf831d8b/master/src/handler.rs:102
at /home/john/.cargo/git/checkouts/rotor-d280b489cf831d8b/master/src/handler.rs:186
at /home/john/.cargo/git/checkouts/mio-361c534895140a24/v0.5.x/src/event_loop.rs:343
at /home/john/.cargo/git/checkouts/mio-361c534895140a24/v0.5.x/src/event_loop.rs:335
at /home/john/.cargo/git/checkouts/mio-361c534895140a24/v0.5.x/src/event_loop.rs:298
26:     0x55fbaab2337f - _<hyper..server..ServerLoop<A, H> as core..ops..Drop>::drop::_{{closure}}::h4215c25b7a8986d8
27:     0x55fbaab06557 - drop::hdde9cbdf94b8878f
28:     0x55fbaab241db - std::panicking::try::_{{closure}}::_{{closure}}::h2d6d2fb605e43bef
29:     0x55fbaaef1afb - __rust_try
30:     0x55fbaaef19de - __rust_maybe_catch_panic
31:     0x55fbaab23e61 - std::thread::Builder::spawn::_{{closure}}::h7cb2f4bd1e31f8a9
32:     0x55fbaab070dd - _<F as alloc..boxed..FnBox<A>>::call_box::h95dfb0e318a6a243
at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/liballoc/boxed.rs:587
33:     0x55fbaaee6554 - std::sys::thread::Thread::new::thread_start::h8f3bd45211e9f5ea
34:     0x7f54eb7f56f9 - start_thread
35:     0x7f54ec29bb5c - clone
36:                0x0 - <unknown>
2016-07-25 19:11:48 Finishing work, please wait...
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Closed', ../src/libcore/result.rs:788
stack backtrace:
1:     0x55fbaaee012f - std::sys::backtrace::tracing::imp::write::h29f5fdb9fc0a7395
2:     0x55fbaaeea2eb - std::panicking::default_hook::_{{closure}}::h2cc84f0378700526
3:     0x55fbaaee7c3a - std::panicking::default_hook::hbbe7fa36a995aca0
4:     0x55fbaaee828c - std::panicking::rust_panic_with_hook::h105c3d42fcd2fb5e
5:     0x55fbaaee80e1 - std::panicking::begin_panic::hbf62ea4a5ff3f9de
6:     0x55fbaaee800a - std::panicking::begin_panic_fmt::h20f5943904e5791d
7:     0x55fbaaee7f7e - rust_begin_unwind
8:     0x55fbaaf2628f - core::panicking::panic_fmt::h19323e466869c656
9:     0x55fbaab2ebe9 - core::result::unwrap_failed::he2c96d4397353866
at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libcore/macros.rs:29
10:     0x55fbaab43932 - hyper::server::Listening::close::h1a0fcc9c4be9b0a2
at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libcore/result.rs:726
at /home/john/.cargo/git/checkouts/hyper-03790fdbcffbe388/master/src/server/mod.rs:318
11:     0x55fbaab1f839 - _<jsonrpc_http_server..Server as core..ops..Drop>::drop::h3f3ef1bd4220155d
at /home/john/.cargo/git/checkouts/jsonrpc-http-server-67dcc044a40796d0/master/src/lib.rs:193
12:     0x55fbaa66d819 - drop::h4f3766c7a449e6c1
13:     0x55fbaa6fa61a - parity::execute_client::hfbb55bd77942235d
at /home/john/code/parity/parity/main.rs:596
at /home/john/code/parity/parity/main.rs:340
14:     0x55fbaa6e95af - parity::execute::h4228f2b3aef67c46
at /home/john/code/parity/parity/main.rs:163
15:     0x55fbaa6e5e2a - parity::main::hc0b8b9a4c77213f8
at /home/john/code/parity/parity/main.rs:113
16:     0x55fbaaee7ee8 - std::panicking::try::call::h5df3ac2979db3c90
17:     0x55fbaaef1afb - __rust_try
18:     0x55fbaaef19de - __rust_maybe_catch_panic
19:     0x55fbaaee71d8 - std::rt::lang_start::hfe9ab243c60ffb9b
20:     0x7f54ec1b582f - __libc_start_main
21:     0x55fbaa5df058 - _start
22:                0x0 - <unknown>
```

**Comment by debris** [comment]
@johnkozan this should be fixed on master in #1698 (it's merged only on master). If you synced with tracing before this pr, you need to resync.

The stack trace looks like you have an old database. I will double check that now. If everything will be fine, I will close the issue

**Comment by mtbitcoin** [comment]
@debris I just did a pull earlier and rebuilt with "20160724". It was still crashing..

**Comment by debris** [comment]
@johnkozan I've checked and the issue does no longer exist

request
```
curl -X POST -H "Content-type: application/json" --data '{"jsonrpc":"2.0","method":"trace_transaction","params":["0x3c29cfb1c194db3034b30876aeb291706dff89f0ff2eed8f247ee25ddcc073aa"],"id":63343}'  localhost:8545
```

response
```
{"jsonrpc":"2.0","result":[{"action":{"call":{"from":"0x03e21763fccbf7e5b56d0d2a461c87320797711f","gas":"0x5a33","input":"0x","to":"0xbb9bc244d798123fde783fcc1c72d3bb8c189413","value":"0x2aed35a700679e00"}},"blockHash":"0xcedced4b4eb5c42f48ebf02f6a645818f911b1da5ff53a413520bf131f7693b5","blockNumber":"0x160f87","result":{"failedCall":[]},"subtraces":"0x00","traceAddress":[],"transactionHash":"0x3c29cfb1c194db3034b30876aeb291706dff89f0ff2eed8f247ee25ddcc073aa","transactionPosition":"0x00"}],"id":63343}
```

@johnkozan @mtbitcoin the issue has been fixed only on latest master. After pulling latest master, you need to remove your database from `~/.parity/906a34e69aec8c0d` directory

**Comment by mtbitcoin** [comment]
@debris so we need to start with a clean DB? Upon startup it states that the database is migrated from version 6 -- 7 if the previous DB is still there. 

Also, i am not sure if this related but i have been having synching problem with the "20160726" build

**Comment by debris** [comment]
> @debris so we need to start with a clean DB? Upon startup it states that the database is migrated from version 6 -- 7 if the previous DB is still there.

Sry. You actually don't have to resync. You can use `export` / `import` features.

```
parity export bc
rm -rf ~/.parity/906a34e69aec8c0d
parity import bc --tracing on
```

> Also, i am not sure if this related but i have been having synching problem with the "20160726" build

It's not related

Also, after #1723 issue is fixed, you may need to repeat `export`/`import` steps. Excuse us that this is manual. We will try to make this improvements automatic for next release

**Comment by mtbitcoin** [comment]
@debris I've opened a new issue (https://github.com/ethcore/parity/issues/1726) for the synch. This only occurred after the upgrade

**Comment by ppratscher** [comment]
I did a full resync today retrieving the trace for this tx still crashes parity (parity version 22a87b1-20160726).

Also now retrieving the trace for block 51921 or 51922 crashes the node but with a different error:
```
thread '<unnamed>' panicked at 'DecodeError: RlpIsTooShort', util/src/rlp/rlpin.rs:119
note: Run with `RUST_BACKTRACE=1` for a backtrace.
2016-07-26 16:08:17 Finishing work, please wait...
thread '<main>' panicked at 'called `Result::unwrap()` on an `Err` value: Closed', ../src/libcore/result.rs:785
```

**Comment by johnkozan** [comment]
@debris I am still experiencing this issue even with latest git master:
```
  version Parity/v1.3.0-unstable-01e33ff-20160726/x86_64-linux-gnu/rustc1.12.0-nightly
```
After compiling, I completely removed parity data with `rm -rf ~/.parity`


During initial sync, parity crashed with:
```
2016-07-26 10:59:36 Syncing #1429444 52f9â€¦16a3    369 blk/s 2556 tx/s  72 Mgas/s    9181+26885 Qed   #1465643   20/26/25 peers     36 MiB db    5 MiB chain   35 MiB queue    2 MiB sync
2016-07-26 10:59:41 Syncing #1431245 458fâ€¦4f79    354 blk/s 1799 tx/s  47 Mgas/s    4099+30232 Qed   #1465643    0/25/25 peers     30 MiB db    9 MiB chain   29 MiB queue    2 MiB sync

thread 'IO Worker #3' has overflowed its stack
fatal runtime error: stack overflow
Aborted (core dumped)
```

After parity was completely synced, the RPC call mentioned previously will crash parity.

backtrace:
```
thread '<unnamed>' panicked at 'DecodeError: RlpIsTooShort', util/src/rlp/rlpin.rs:119
stack backtrace:
1:     0x55fbaaee012f - std::sys::backtrace::tracing::imp::write::h29f5fdb9fc0a7395
2:     0x55fbaaeea2eb - std::panicking::default_hook::_{{closure}}::h2cc84f0378700526
3:     0x55fbaaee7c3a - std::panicking::default_hook::hbbe7fa36a995aca0
4:     0x55fbaaee828c - std::panicking::rust_panic_with_hook::h105c3d42fcd2fb5e
5:     0x55fbaaee80e1 - std::panicking::begin_panic::hbf62ea4a5ff3f9de
6:     0x55fbaaee800a - std::panicking::begin_panic_fmt::h20f5943904e5791d
7:     0x55fbaaabd920 - _<ethcore_util..kvdb..Database as ethcore..db..Readable>::read::_{{closure}}::h601b7564885109ad
at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libcore/result.rs:695
at /home/john/code/parity/util/src/rlp/rlpin.rs:119
at /home/john/code/parity/util/src/rlp/rlpin.rs:124
at /home/john/code/parity/util/src/rlp/mod.rs:90
at /home/john/code/parity/ethcore/src/db.rs:164
8:     0x55fbaaa17f43 - _<ethcore..trace..db..TraceDB<T>>::traces::h5a7172974d0ddb60
...
```

**Comment by ppratscher** [comment]
I can confirm the issue as well. Encountered it while testing the _trace-other-calls_ branch.

**Comment by ppratscher** [comment]
Thanks for resolving this issue! Will it require a full  chain resume?

**Comment by mtbitcoin** [comment]
For reference i am running into slightly different issue now https://github.com/ethcore/parity/issues/1760

**Comment by gavofyork** [comment]
not sure; @debris ?
